torch_lr_finder/lr_finder.py

```diff
@@ -186,17 +249,31 @@ def range_test(
             raise ValueError("smooth_f is outside the range [0, 1[")
 
         # Create an iterator to get data batch by batch
-        iter_wrapper = DataLoaderIterWrapper(train_loader)
+        if isinstance(train_loader, DataLoader):
+            train_iter = TrainDataLoaderIter(train_loader)
+        elif isinstance(train_loader, TrainDataLoaderIter):
+            train_iter = train_loader
+        else:
+            raise ValueError("`train_loader` has unsupported type: {}".format(type(train_loader)))
+        
+        if val_loader:
+            if isinstance(val_loader, DataLoader):
+                val_iter = ValDataLoaderIter(val_loader)
+            elif isinstance(val_loader, ValDataLoaderIter):
+                val_iter = val_loader
+            else:
+                raise ValueError("`val_loader` has unsupported type: {}".format(type(val_loader)))
```

Comment:
We should let the user know which types are expected: Dataloader and ValDataLoaderIter